### PR TITLE
Add account workspace setting type

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
Add two new types to `settings/generic_setting.go` - `accountWorkspaceSettingDefinition` and `accountWorkspaceSetting`, to allow settings that can be set at both workspace & account level
This may simplify PRs like #3325

## Tests

- [ ] `make test` run locally
- [ ] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: settings/generic_setting.go [hunk 1]
```diff
@@ -199,6 +199,100 @@ func (w accountSetting[T]) GetId(t *T) string {
...
+
+	// Optional function to generate resource ID from the settings. If not provided, will use predefined value `global`
+	generateIdFunc func(setting *T) string
+}
+
+func (aw accountWorkspaceSetting[T]) SettingStruct() T {
+	return aw.settingStruct
+}
+func (aw accountWorkspaceSetting[T]) Read(ctx context.Context, c *common.DatabricksClient, etag string) (*T, error) {
+	if c.Config.IsAccountClient() {
```
Comment (mgyucht) [inline settings/generic_setting.go:235]: We can use `c.AccountOrWorksapceRequest` for all of these.

File: settings/generic_setting.go [hunk 2]
```diff
@@ -149,7 +149,7 @@ var _ workspaceSettingDefinition[struct{}] = workspaceSetting[struct{}]{}
 
 type accountSettingDefinition[T any] genericSettingDefinition[T, *databricks.AccountClient]
 
-// An account setting is a setting that is scoped to a workspace.
+// An account setting is a setting that is scoped to an account.
```
Comment (mgyucht) [inline settings/generic_setting.go:152]: Oops :)

**Review by mgyucht — APPROVED** [review]
LGTM, but suggest to use `c.AccountOrWorkspaceRequest` to minimize duplicating logic.

File: settings/generic_setting.go (hunk 1 above)
Comment (nkvuong) [reply settings/generic_setting.go:235]: `AccountOrWorkspaceRequest` is a method on `DatabricksClient`, so won't accept a type parameter 😢

**Review by tanmay-db — APPROVED** [review]

**Pull request merged**